ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape.cpp

```diff
@@ -64,7 +64,7 @@ ttnn::Tensor row_major_reshape(const ttnn::Tensor& tensor, const ttnn::Shape& sh
         if (rm_tensor.is_contiguous()) {
             // Page size depends on the width, so only modify the shape if the width is the same
             if (tensor_shape_with_padding[-1] == shape_with_padding[-1]) {
-                return rm_tensor.reshape(shape);
+                reshaped_rm_tensor =  rm_tensor.reshape(shape);
```

Comment:
Do I get it right that function `row_major_reshape` might accept Tiled input and product a Tiled output? 🤔  
A bit confusing

Reply:
It describes that the reshape is being done in row major (to remove the padding) but after it puts it back to the same layout as the original input.

Reply:
Yeah, I get it. The name is confusing. 
what exact unique cases it supports that other forms of reshape don’t? The name could tell 
* or what type it handles (not the case here)
* what cases it handles (better)
* or how it handles (least desirable, but can be ok. this is what this name here tries to do but fails imho, because it’s not clear that row_major in the name refers to the conversion to row major that takes place internally (optionally?))


So the feedback here is to please think of a proper name. 
not a blocker for the PR.

Reply:
I modified the name, hopefully its better :)